### PR TITLE
Alerts slack integration UI

### PR DESCRIPTION
**Pull request by AdityaHegde**
closes [#4410](https://github.com/rilldata/rill/issues/4410)

Adds a section to add slack notifications if it is enabled.

Run `rill env set "connector.slack.bot_token" "xoxb-..." --project ...` to enable slack connector.

### CODE REVIEW

**Comment by ericpgreen2** [comment]
I'm seeing the Slack section, even though I haven't yet enabled the connector.

image

File: web-common/src/components/forms/FormSection.svelte [hunk 1]
```diff
@@ -1,11 +1,14 @@
 <script lang="ts">
+  import Switch from "@rilldata/web-common/components/forms/Switch.svelte";
   import InfoCircle from "@rilldata/web-common/components/icons/InfoCircle.svelte";
   import Tooltip from "@rilldata/web-common/components/tooltip/Tooltip.svelte";
   import TooltipContent from "@rilldata/web-common/components/tooltip/TooltipContent.svelte";
 
   export let title: string;
   export let description: string = "";
   export let padding = "p-3";
+  export let showEnabled = false;
```
Comment (ericpgreen2) [inline web-common/src/components/forms/FormSection.svelte:10]: Name suggestion: `showSectionToggle`

File: web-common/src/features/alerts/BaseAlertForm.svelte [hunk 2]
```diff
@@ -2,13 +2,18 @@
...
   import AlertDialogDeliveryTab from "./delivery-tab/AlertDialogDeliveryTab.svelte";
-  import { checkIsTabValid, FieldsByTab } from "./form-utils";
+  import {
+    type AlertFormValues,
+    checkIsTabValid,
+    FieldsByTab,
+  } from "./form-utils";
 
-  export let formState: any; // svelte-forms-lib's FormState
+  export let formState: ReturnType<typeof createForm<AlertFormValues>>;
```
Comment (ericpgreen2) [inline web-common/src/features/alerts/BaseAlertForm.svelte:16]: Great to see this typing 👍

**Review by ericpgreen2 — APPROVED** [review]
Code looks good 👍

**Comment by AdityaHegde** [comment]
Good catch about the slack showing up. There was a bug in how i used the signal.

**Pull request merged**